blivet/__init__.py

```diff
@@ -873,7 +873,9 @@ def clearPartitions(self):
             if should_clear:
                 self.recursiveRemove(disk)
 
-            if zerombr or should_clear:
+            if disk.protected:
+                log.warning("cannot clear '%s': disk is protected or read only", disk.name)
```

Comment:
If you placed this check in the `elif zerombr or should_clear:` branch it would only appear in the logs when relevant, ie: not when the read-only drive was properly excluded via `ignoredisk`.

Reply:
Fixed.